ui/store/actions.ts

```diff
@@ -1136,6 +1137,22 @@ export function enableSnap(
   };
 }
 
+export function updateSnap(
+  origin: string,
+  snap: { [snapId: string]: { version: string } },
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return async (dispatch: MetaMaskReduxDispatch, getState) => {
+    submitRequestToBackground('installSnap', [origin, snap]);
+    await forceUpdateMetamaskState(dispatch);
```

Comment:
`forceUpdateMetamaskState` returns the new state, so I think we can simplify this

Reply:
If I do so, the UX becomes broken again. So, the popup with the update screen will remain in the background. I think that @GuillaumeRx have strong reasons why this is all implemented like that.

Reply:
Yeah, you are right. I think this has a race condition. We may want to find a way to wait for the approval to trigger. Currently I think we could be unlucky at get stuck.